lib/seahorn/HornCex.cc

```diff
@@ -373,5 +386,62 @@ namespace seahorn
     out.keep ();
   }
 
+  static void sliceErrorTrace(Function &F, BmcTrace &trace) {
```

Comment:
I was aware of the code-reuse issue here. However, sliceErrorTrace has different semantics than reduceToRegion, for example, reduceToRegion assumes the input is a precise region while BmcTrace may omit some basic blocks due to abstraction. Not sure of the impact on changing the semantics of reduceToRegion directly.